plugins/spectrum_laser_protection.py

```diff
@@ -0,0 +1,151 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 7 March 2025
+
+@author: Éric Piel
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License version 2 as published by the Free
+Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+# Changes the behaviour of the GUI, so that on a SPARC, it's not possible to turn on the laser
+# of a spectrum stream when the parabolic mirror is not engaged. The idea is that, although unlikely,
+# the user could go to the "acquisition" tab without having engaged the mirror (typically, it's not useful,
+# but the GUI allows this). In this case, the input laser (coming from via a FPLM module) could enter
+# the chamber and hit and damage some of the detectors.
+# The way it works in practice is that whenever the acquisition tab is shown, it checks the parabolic mirror
+# position. If it's not engaged it:
+# * sets the spectrum stream light power to 0, and disable the entry. So it's impossible to turn it on.
+# * when a new spectrum stream is added, it disables the power entry.
+# If the mirror is engaged, the power entry of all spectrum streams is re-enabled.
+# For testing, use the sparc2-fplm-sim.odm.yaml microscope file.
+
+import functools
+import logging
+from typing import Dict, Callable
+
+from odemis.acq.stream import SpectrumSettingsStream
+from odemis.gui.cont.stream import StreamController
+from odemis.gui.cont.tabs import MIRROR_ENGAGED
+from odemis.gui.plugin import Plugin
+
+
+class SpectrumLaserProtectionPlugin(Plugin):
+    name = "Forbid to turn on laser during spectrum acquisition if parabolic mirror is not engaged"
+    __version__ = "1.0"
+    __author__ = "Éric Piel"
+    __license__ = "GPLv2"
+
+    def __init__(self, microscope, main_app):
+        super().__init__(microscope, main_app)
+        # Can only be used with a SPARC with spectrometer(s)
+        main_data = self.main_app.main_data
+        if microscope and main_data.role.startswith("sparc"):
+            self._tab = main_data.getTabByName("sparc_acqui")
```

Comment:
for readability in `on_tab_change`, I would suggest to rename `_tab` to `_acqui_tab`
```suggestion
            self._acqui_tab = main_data.getTabByName("sparc_acqui")
```